spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java

```diff
@@ -994,6 +1021,22 @@ void customizeJetty(ServerProperties serverProperties,
 			if (serverProperties.getMaxHttpPostSize() > 0) {
 				customizeMaxHttpPostSize(factory, serverProperties.getMaxHttpPostSize());
 			}
+
+			customizeConnectionTimeout(serverProperties, factory);
+		}
+		private void customizeConnectionTimeout(final ServerProperties serverProperties,
+													final JettyEmbeddedServletContainerFactory factory) {
+			factory.addServerCustomizers(new JettyServerCustomizer() {
+				@Override
+				public void customize(final Server server) {
+					for (org.eclipse.jetty.server.Connector connector : server.getConnectors()) {
+						if (connector instanceof ServerConnector) {
+							ServerConnector serverConnector = (ServerConnector) connector;
+							serverConnector.setSoLingerTime(serverProperties.getConnectionTimeout());
```

Comment:
This should be a call to `setIdleTimeout` instead. It is described as "sets the maximum Idle time for a connection, which roughly translates to the `Socket.setSoTimeout(int)` call.